src/node/internal/internal_zlib.ts

```diff
@@ -20,6 +23,253 @@ const {
   CONST_UNZIP,
 } = zlibUtil;
 
+export function crc32(
+  data: ArrayBufferView | string,
+  value: number = 0
+): number {
+  validateUint32(value, 'value');
+  return zlibUtil.crc32(data, value);
+}
+
+export function inflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATE));
+}
+
+export function deflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATE));
+}
+
+export function gunzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GUNZIP));
+}
+
+export function gzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GZIP));
+}
+
+export function inflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATERAW)
+  );
+}
+
+export function deflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATERAW)
+  );
+}
+
+export function unzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATERAW)
+  );
+}
+
+function normalizeArgs(
+  optionsOrCallback: ZlibOptions | CompressCallback<Error, Buffer>,
+  callbackOrUndefined?: CompressCallback<Error, Buffer>
+): [ZlibOptions, CompressCallback<Error, Buffer>] {
+  if (typeof optionsOrCallback === 'function') {
+    return [{}, optionsOrCallback];
+  } else if (typeof callbackOrUndefined === 'function') {
+    return [optionsOrCallback, callbackOrUndefined];
+  }
+
+  throw new ERR_INVALID_ARG_TYPE('callback', 'Function', callbackOrUndefined);
+}
+
+function wrapCallback(
+  callback: CompressCallback<Error, Buffer>
+): CompressCallback<string, ArrayBuffer> {
+  return (error: string | null, result: ArrayBuffer | null) => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    queueMicrotask(() => {
+      callback(
+        error ? new Error(error) : null,
+        result ? Buffer.from(result) : Buffer.alloc(0)
```

Comment:
Shouldn't second parameter be null, if there is an error?

Reply:
It should be undefined in Node. However, DefinitelyTyped type says it's always Buffer. Not sure what to do with this mess.

Reply:
I would make it `undefined` then.